### ISSUE TITLE
Decide Data Model

### ISSUE THREAD

**Issue by barbados-clemens** [barbados-clemens/shop-the-fridge#15]

**Comment by barbados-clemens** [comment]
The base unit should be an _item_ or _ingredient_ 
an ingredient can have basic properties like 

## Item
- id
    - type number
- name
    - type string
- expiresAt
    - type  Date
- thumbnail
    - type string 
    - URL to user desired image
- quantity
    - Type number 
    - something like 1 [unit] 
- unitType
    - type string 
    - something like cup/oz/can/ etc probably an enum. with quantity and unit you will get 1 cup or 1 can etc.
- createdAt
    - type Date 
- updatedAt
    - type Date 
```

A list of Items is a Recipe or something 

## Recipe
- id
    - type number
- name
    - type string
    - name of dish like _green bean casserole_
- ingredients
    - type Item/Ingredient array 
- createdAt
    - type Date
- steps
    - type string array
    - ordered list of instructions
- link
    - type string
    - used for linking out to recipe or other resource the user would like to remember about this recipe


A list of Recipes is a Plate or Meal


# Plate
- id
    - type number
- recipes
    - type recipe array
- createdAt
    - type Date


We'd also need a shopping list which would be a list of ingredients

That's at least my thinking initially 

What y'all think @karnthis @Woddell?

**Comment by karnthis** [comment]
I personally would prefer if we had a separate table of possible ingredients. The FDA has a database of every product and it's barcode, I was thinking we could have the master list, and then each user would have a reference table of what they had added linking to the master ingredients.

More thoughts to come.

**Comment by barbados-clemens** [comment]
I agree having a seperate ingredient table, then if a user wants to add an amount or something then it's added to their own "table" or something with an id back to the ingredients. 

FDA table which I think is a good idea. though still be able to add to that table as there are some local grocery store things where they don't have official barcodes or only in house items.

**Comment by barbados-clemens** [comment]
Here are the data models I've been working with so far on the frontend just to prototype out the UI. 
they are still changing as I find what properties I need to perform certain tasks. 

https://github.com/barbados-clemens/shop-the-fridge/tree/feature/recipe-ui/apps/web/src/app/models
